src/ui/modals/smithing-modal-window.js

```diff
@@ -0,0 +1,156 @@
+import { FONTS } from "../../constants/constants.js";
+import { ModalWindow } from "../modal-window.js"
+import { Button } from "../button.js";
+import { getItemClass } from "../../utilities.js";
+import { itemManifest } from "../../items/item-manifest.js";
+
+export class SmithingModalWindow extends ModalWindow {
+    x = 10;
+    y = 80;
+
+    visible = false;
+
+    window;
+    elements = [];
+    elementEvents = [];
+    choice = "None";
+
+    leftOffset = 8;
+    topOffset = 32;
+
+    horizontalPadding = 8;
+    verticalPadding = 8;
+
+    iconWidth = 60;
+    iconHeight = 32;
+    buttonWidth = 48;
+    buttonHeight = 48;
+
+    maxIconsPerRow = 5;
+
+    constructor(scene) {
+        super();
+        this.scene = scene;
+
+        this.window = this.scene.add
+            .image(this.x, this.y, "smithing-interface")
+            .setOrigin(0, 0)
+            .setDepth(2);
+        
+        this.window.visible = this.visible;
+
+        this.exitButton = new Button(this.scene, this.x + 473, this.y + 8, 19, 19);
+        this.exitButton.on("pointerup", () => {
+            if (this.visible) {
+                this.setVisible(false);
+            }
+        });
+    }
+
+    clearChoices() {
+        for (let element of this.elements) {
+            element.destroy();
+        }
+        
+        this.elements = [];
+    }
+
+    getSpriteName(name) {
+        let result = name.toLowerCase();
+        result.toLowerCase();
+        result = result.replaceAll(" ", "-");
+        return result;
+    }
+
+    async setChoices(bar, barSupply, smithingLevel) {
+        this.clearChoices();
+
+        let elements = [];
+
+        switch (bar) {
+            case "Bronze Bar":
+                    elements = ["BronzeDagger", "BronzeSword", "SKIP", "BronzeScimitar", "BronzeAxe", "Bronze2hSword"];
+                break;
+            default:
+                return;
+        }
+
+        let rowIndex = 0;
+
+        let positionX = this.x + this.leftOffset + this.horizontalPadding;
+        let positionY = this.y + this.topOffset + this.verticalPadding;
+
+        for (let element of elements) {
+            if (element === "SKIP") {
+                positionX += this.iconWidth + this.horizontalPadding;
+                rowIndex += 1;
+                continue;
+            }
+
+            if (rowIndex >= this.maxIconsPerRow) {
+                positionX = this.x + this.leftOffset + this.horizontalPadding;
+                positionY += this.iconHeight + this.verticalPadding + 24;
```

Comment:
In general for loop creation logic like this I like to create these variables based off the index rather than adding to them.
Like instead of  `+=`  do:

```js
const yOffset = this.iconHeight + this.verticalPadding + 24;
const xOffset = ...
for (let [index, val] of array.entries()) {
    const positionY = index * yOffset;
    const positionX = index * xOffset;
    // ...
}
```

@EthanVieira Thoughts?

Another potential way of making this looping creation easier to read is mapping over the data and returning that into `elements` instead of creating elements and appending to it

Reply:
I don't think I have a preference. Combining the terms into one offset does help readability though